core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -153,7 +156,7 @@ class SearchService(
   def refreshResolver(): Unit = resolver.update()
 
   def persist(check: FileCheck, symbols: List[FqnSymbol], commitIndex: Boolean, boost: Boolean): Future[Option[Int]] = {
-    val iwork = Future { blocking { index.persist(check, symbols, commitIndex, boost) } }
+    val iwork = Future { blocking { index.persist(check, symbols, commitIndex, boost) } }(ex)
```

Comment:
I seem to remember something about us having some profiler licenses kicking
around, is that still true? And if so, may I use one?
On May 2, 2016 8:09 AM, "Sam Halliday" <notifications@github.com> wrote:

> In core/src/main/scala/org/ensime/indexer/SearchService.scala
> <https://github.com/ensime/ensime-server/pull/1436#discussion_r61731085>:
>
> > @@ -153,7 +156,7 @@ class SearchService(
> >    def refreshResolver(): Unit = resolver.update()
> >
> >    def persist(check: FileCheck, symbols: List[FqnSymbol], commitIndex: Boolean, boost: Boolean): Future[Option[Int]] = {
> > -    val iwork = Future { blocking { index.persist(check, symbols, commitIndex, boost) } }
> > +    val iwork = Future { blocking { index.persist(check, symbols, commitIndex, boost) } }(ex)
>
> this should not be bounded. The thing we need to back pressure is the
> indexing (which generates the memory overhead), not the persisting (which
> clears the heap).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ensime/ensime-server/pull/1436/files/9cdfbe3f20f47d617f0fd30471f8a08630f81a40#r61731085>
>